server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -615,7 +615,7 @@ private static Object deepCopy(Object value) {
             return ((Date) value).clone();
         } else if (value instanceof ZonedDateTime) {
             ZonedDateTime zonedDateTime = (ZonedDateTime) value;
-            return ZonedDateTime.of(zonedDateTime.toLocalDate(), zonedDateTime.toLocalTime(), zonedDateTime.getZone());
+            return ZonedDateTime.ofInstant(zonedDateTime.toLocalDateTime(), zonedDateTime.getOffset(), zonedDateTime.getZone());
```

Comment:
TBH I'm wondering whether this could just use `zonedDateTime` directly, since it is supposed to have value semantics. Put differently, why not also deep-copy the `LocalDateTime` and `ZoneId` and so on?

Reply:
I'm not sure I understand. does it matter?

Reply:
It _does_ but I should have added a test to show this. It only very rarely matters, which is why it's trappy. I added a test.

Reply:
Hmm, ok, I forgot I'd made that original comment, but I think I was answering a different question in my previous reply.

The question is why not simply this:

```java
else if (value instanceof ZonedDateTime) {
    return value;
```

I thought this would be ok because `ZonedDateTime` is immutable and we shouldn't be looking at reference equality on it. If that's _not_ fine, because we do want the contents of the copy to share no references with the original for some reason, then what we do here doesn't do that since it makes a shallow copy of the `ZonedDateTime`, because it copies the `LocalDateTime` instance within.

Reply:
++ on that